server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadataVerifier.java

```diff
@@ -74,6 +74,10 @@ public IndexMetadata verifyIndexMetadata(IndexMetadata indexMetadata, Version mi
 
         // First convert any shared_cache searchable snapshot indices to only use _tier_preference: data_frozen
         IndexMetadata newMetadata = convertSharedCacheTierPreference(indexMetadata);
+        // Remove _tier preference settings if available, because though these are technically not
```

Comment:
```suggestion
        // Remove _tier routing settings if available, because though these are technically not
```

Reply:
Sorry about this nit, but I got tripped a bit as `_tier_preference` is already a setting name (and it's not the one we're removing)